src/lib/storage/run_length_segment/run_length_segment_iterable.hpp

```diff
@@ -44,143 +43,192 @@ class RunLengthSegmentIterable : public PointAccessibleSegmentIterable<RunLength
  private:
   const RunLengthSegment<T>& _segment;
 
+  /**
+   * Due to the nature of Run Length encoding, point-access is not in O(1).
+   * Run length segments store the end positions of runs in a sorted vector. To access a particular position, this
+   * vector has to be searched for the end position of the according run. It is possible to find the value of a
+   * position in O(log(n)) by doing a binary search.
+   * In case we have already searched for a nearby position (e.g., when a previous not selective filter removed only
+   * relatively few lines), we can linearly search the end position. Since position lists are often ordered and modern
```

Comment:
Minor: previously, search *for*

Reply:
Where would you use `previously` over `previous`? Or did I get that wrong?
Added the `for`.

Reply:
Gnah, I am not happy with the quality of my review. :/

Actually, I believe it should be "when a previous **,** non-selective". Or, maybe better: "a previous filter with a low selectivity resulted in an almost contiguous PosList"?

Reply:
Like the last one. Done.